sys/aarch64/tlb.c

```diff
@@ -6,13 +6,13 @@
 #define __dsb(x) __asm__ volatile("DSB " x)
 #define __isb() __asm__ volatile("ISB")
 
-void tlb_invalidate(pte_t pte, asid_t asid) {
+void tlb_invalidate(vaddr_t va, asid_t asid) {
   __dsb("ishst");
 
   if (asid > 0) {
-    __tlbi("vae1is", ASID_TO_PTE(asid) | (pte >> PAGE_SHIFT));
+    __tlbi("vae1is", ASID_TO_PTE(asid) | (va >> PAGE_SHIFT));
```

Comment:
Umm... can you explain what is the difference between `vae1is` and `vaae1is`, or at least point a reader to documentation.

Reply:
`vae1is` - Invalidate translation used at EL1 for the specified VA and Address Space Identifier (ASID) and the current VMID, Inner Shareable.
`vaae1is` - Invalidate all translations used at EL1 for the specified address and current VMID and for all ASID values, Inner Shareable.

Based on [arm documentation](https://developer.arm.com/documentation/ddi0488/h/system-control/aarch64-register-summary/aarch64-tlb-maintenance-operations) and `pmap_invalidate_page` from FreeBSD - `/sys/arm64/arm64/pmap.c`

Reply:
Please put it in a comment it the source code.

Reply:
done

Reply:
IIUC `asid = 0` only for kernel address space, and only kernel can use addresses that are translated with PT pointed by TTBR1. Why do we need to use `vaae1is` for kernel AS, instead of just `vae1s` ?

Reply:
I don't know answer for your question. _NetBSD_, _FreeBSD_, _Linux_ and even _Fuchsia_ use `vaae1is` for kernel.
After removing `vaae1is` branch everything still works but I don't want to leave potential bug in that part of code for future.

Reply:
Ok. As usual - please put your findings into a comment in the code.

Reply:
done